### PR TITLE
Upgrade to Vite 4

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

Upgrade to Vite 4, along with Vite plugins that has peer on Vite 4.

Also updated the Svelte integration relying on `vite-plugin-svelte` v2 features, which should simplify it's configuration.

## Testing

Updated some test utils to work in Vite 4. Existing tests should pass.

## Docs

I suppose we would document the migration guide once we're near a stable release? cc @withastro/maintainers-docs

### CODE REVIEW

File: packages/astro/test/test-utils.js
Comment (bluwy) [inline packages/astro/test/test-utils.js:143]: Vite 4 doesn't mutate `NODE_ENV` anymore, unless it's `undefined`. So if we run the dev server, and then build, build would run in `development` `NODE_ENV` instead, which broke the MDX test.

This shouldn't affect users in practice, but internally for our tests we need to make sure `NODE_ENV` is the right value we're testing.

File: packages/integrations/svelte/src/index.ts
Comment (bluwy) [inline packages/integrations/svelte/src/index.ts:24]: `vitePreprocess` does the same thing as `svelte-preprocess`, except it's relying on Vite's internal processing like we do, so it should be slightly more performant.

This introduced a small breaking change where `lang="postcss"` is required for `vitePreprocess` to process to minimize preprocess overhead, noted in the changesets.

File: packages/integrations/svelte/src/index.ts
Comment (bluwy) [inline packages/integrations/svelte/src/index.ts:68]: vite-plugin-svelte includes the removed identifiers by default.

File: packages/integrations/svelte/src/index.ts
Comment (bluwy) [inline packages/integrations/svelte/src/index.ts:87]: This removal is safe because internally Astro passes it down to Vite, which when we use `vitePreprocess`, it would include this configuration.

**Comment by delucis** [comment]
> I suppose we would document the migration guide once we're near a stable release? cc @withastro/maintainers-docs

Probably! I think I remember you saying this is mostly non-breaking? Do we have a quick summary of what Astro users are most likely to run into? Or should everyone read the Vite migration doc?

**Comment by bluwy** [comment]
I think we might want users to skim through https://vitejs.dev/guide/migration.html first as it's not a lot. But I'd say the most likely breaking change someone might hit is incompatible Vite/Rollup plugins (Rollup 3).

The changeset notes could probably be enough, and I bet writing good ones there would ease writing the migration guide later. So perhaps I do need docs maintainers to review the changeset 🤔

---

PS I updated some tests but I'm expecting the smoke tests to fail still. Will need to dig that deeper.

**Comment by bluwy** [comment]
Marking as ready for review. Smoke test should still fail as it's blocked by https://github.com/withastro/astro/pull/5706

**Review by matthewp — APPROVED** [review]

**Review by delucis — APPROVED** [review]
Read through the Vite 4 migration guide and I think @bluwy’s right we don’t necessarily need to call out too much in terms of Astro migration for this one. So no more docs than the changesets needed for now I think.

**Pull request merged**
